### DISCUSSION TITLE
feature request: deterministic key generation

### DISCUSSION THREAD

**Discussion by sneak** [FiloSottile/age#393] [General]
I'd like to generate a pseudoinfinite list of deterministic private age keys, ideally from a leaf of a BIP32-style seed phrase.

This is a feature request for `age-keygen` to support either:

- the use of a bip32 seed phrase and a derivation path to deterministically generate a private key

or

- the use of an "entropy file" as input to `age-keygen` instead of platform-specific random sources for generic deterministic key generation

**Comment by keisentraut** [comment]
> I'd like to generate a pseudoinfinite list of deterministic private age keys, ideally from a leaf of a BIP32-style seed phrase.

I had the need to generate a deterministic key from a passphrase, too. So I wrote a little utility a week ago and just updated it for your usecase of generating more than one secret key. I did not bother implementing BIP32 derivation paths, but you can use any passphrase and generate up to 2^64 secret keys from it.

Please take a look at https://github.com/keisentraut/age-keygen-deterministic but use at your own risk.

**Reply by sylvorg** [reply]
Sorry for bothering you here, but I was wondering how the functionality of `age-keygen-deterministic` could be recreated using the new `batchpass` plugin? I can't seem to find a way to recreate an identity using a passphrase, in case I don't have access to my Yubikeys on a particular system.
